### ISSUE TITLE
Gmsh: Support for PhysicalNames section

### ISSUE THREAD

**Issue by dalcinl** [nschloe/meshio#149]
I would like to add support for writing `$PhysicalNames` sections:

```
$PhysicalNames
number-of-names
physical-dimension physical-number "physical-name"
…
$EndPhysicalNames
```

However, the `read()` function returns `field_data` missing the `physical-dimension` information. I'm not sure what to do here regarding backward compatibility. Right now `field_data` is a dict `{"physical-name": physical-number, ...}`. 

* We can either make a backward incompatible change to output `{"physical-name": (physical-dimension, physical-number), ...}`.

* Or append some sort or private-looking special key, let say `__gmsh@physical_dimension` mapping to a dict `{"physical-name": physical-dimension, ...}`.

Just take a decision according to your preference, and I'll contribute the PR.

PS: My use case is high order meshes. I generate high order meshes with Gmsh, then I use `meshio` to read it, extract the low-order mesh (i.e just the topology or skeleton of the mesh) and write it back to disk alongside with the high-order coordinates of each cell. Then I can feed a high-order DG-like code with these two pieces of information. However, right now I have no easy way to write back the `$PhysicalNames` section, as the `read()` function does not return the full info.

**Comment by nschloe** [comment]
I wouldn't be afraid to break backwards compatibility, after all we can bump the major version. 

> However, the read() function returns field_data missing the physical-dimension information.

Wait, isn't that information contained in the array itself? I.e., if you have a `400x3` array, the physical dimension is 3?

**Comment by dalcinl** [comment]
What array are you talking about? The `$PhysicalNames` section does not contain arrays, just integers and strings.

Look at the example below. The line `2 1 "outer"`  means that `"outer"` is the physical name of a surface labeled with `1`, while `3 5 "domain"` corresponds to a volume labeled with `5`. 

* Input `.geo` file
```
...
Physical Volume  ("domain",5) = {domain[]};
Physical Surface ("outer",1) = {outer[]};
Physical Surface ("inner",2) = {inner[]};
...
```

* Output ASCII `.msh` file
```
$MeshFormat
2.2 0 8
$EndMeshFormat
$PhysicalNames
3
2 1 "outer"
2 2 "inner"
3 5 "domain"
$EndPhysicalNames
...
```

If backward compatibility is not an issue, then I propose that `meshio` should return a `field_data` dict like this `{"outer": (1, 2), "inner": (2,2), "domain": (3, 5)}`, i.e. `key: "physical-name"`, `value: (physical-number, physical-dimension)`.

Are you ok with this proposal?

**Comment by nschloe** [comment]
I see. Well, field data is a tricky thing since every mesh format has their own set of "extra" data in some form. Typically those would be strings with arrays. I'd propose instead of tuples (`(1, 2)`), you take arrays (`numpy.array([1, 2])`), perhaps with an enforced `dtype=int`. Those should be seamlessly readable and writable by other formats as well, and it would be nice if this info doesn't get lost in format translation.

The only thing that you have to take care of then is in `gmsh_io.write()` to only admit field arrays of type `int` and length `2`, and to issue a warning if you're skipping a field.

**Comment by nschloe** [comment]
Fixed by #150.
